server/pairing/common.go

```diff
@@ -256,22 +262,94 @@ func setDefaultNodeConfig(c *params.NodeConfig) error {
 		return err
 	}
 
-	if len(c.Networks) == 0 {
+	// following specifiedXXX variables are used to check if frontend has specified the value
+	// if not, the default value is set. NOTE: we also check 2 extra fields: WakuV2Config(LightClient|Nameserver)
+	// see api.SetFleet for more details
+	specifiedVerifyTransactionURL := c.ShhextConfig.VerifyTransactionURL
+	specifiedVerifyENSURL := c.ShhextConfig.VerifyENSURL
+	specifiedVerifyENSContractAddress := c.ShhextConfig.VerifyENSContractAddress
+	specifiedVerifyTransactionChainID := c.ShhextConfig.VerifyTransactionChainID
+	specifiedNetworkID := c.NetworkID
+	specifiedNetworks := c.Networks
+	specifiedUpstreamConfigURL := c.UpstreamConfig.URL
+	specifiedLogEnabled := c.LogEnabled
+	specifiedLogLevel := c.LogLevel
+	specifiedFleet := c.ClusterConfig.Fleet
+	specifiedInstallationID := c.ShhextConfig.InstallationID
+	specifiedTorrentConfigEnabled := c.TorrentConfig.Enabled
+	specifiedTorrentConfigPort := c.TorrentConfig.Port
+
+	if len(specifiedNetworks) == 0 {
```

Comment:
Maybe I'm missing something, but this looks very confusing to me.
I know you have plans on re-implementing this with smth like `login-with-syncing` request concept. I think we need to go with without merging this PR. What's the point?

We really should not pass `nodeConfig` from client side, this is wrong. And will very soon become a pain, as we want to remove `nodeConfig` handling from desktop.

-----

1. In general there's no need on passing `NodeConfig` from client.
In desktop we send a very default struct, no custom fields.

2. If we look at `ReceiverConfig`, we can see a few customizations there already, e.g. `SettingCurrentNetwork`:
https://github.com/status-im/status-go/blob/3bacb84a02406da295406acb9d8a9f9c82a25808/server/pairing/config.go#L25-L42

Let's just follow this path and add required fields to this structure. 
For sake of simplicity/readability/maintainability let's compose it into a single struct, maybe `NodeConfigOptions` (or other better name):
```go
type NodeConfigOptions struct { // find better naming
  VerifyTransactionURL string
  ...
}
```

-----

@qfrank let me know what you think
We could merge this solution and improve it in a separate PR, but I don't understand why not do it properly straightaway.